controllers/sr_controller/sr/robot/ruggeduino.py

```diff
@@ -17,36 +17,63 @@ def init_ruggeduino_array(webot):
         "left finger sensor",
         "right finger sensor",
     ]
+    led_names = [
+        "led 1",
+        "led 2",
+        "led 3",
+        "led 4",
+        "led 5",
+        "led 6",
+    ]
+
+    analogue_input_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
 
-    analogue_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
+    digital_input_array = [Microswitch(webot, name) for name in switch_names]
 
-    digital_array = [Microswitch(webot, name) for name in switch_names]
+    digital_output_array = [Led(webot, name) for name in led_names]
 
-    return [Ruggeduino(webot, analogue_array, digital_array)]
+    return [Ruggeduino(webot, analogue_input_array, digital_input_array, digital_output_array)]
 
 
 class Ruggeduino(object):
 
-    DIGITAL_PIN_OFFSET = 2  # Exclude pins 0 and 1 as they are used for USB serial comms
+    DIGITAL_READ_PIN_START = 2  # Exclude pins 0 and 1 as they are used for USB serial comms
+    DIGITAL_WRITE_PIN_START = 7   # Offset output pins by input pins
+    DIGITAL_END_PIN = 12
 
-    def __init__(self, webot, analogue_array, digital_array):
+    def __init__(self, webot, analogue_input_array, digital_input_array, digital_output_array):
         self.webot = webot
-        self.analogue_array = analogue_array
-        self.digital_array = digital_array
+        self.analogue_input_array = analogue_input_array
+        self.digital_input_array = digital_input_array
+        self.digital_output_array = digital_output_array
 
     def digital_read(self, pin):
         "Read an digital input"
-        if pin < 2 or pin > 13:
-            raise ValueError("Only pins 2 - 13 are available on the Ruggeduino")
-        return self.digital_array[pin - Ruggeduino.DIGITAL_PIN_OFFSET].read_value()
+        min_pin = Ruggeduino.DIGITAL_READ_PIN_START
+        max_pin = Ruggeduino.DIGITAL_WRITE_PIN_START - 1
```

Comment:
If we're allowing users to read the LED state back the digital read range needs to be changed:

```suggestion
        max_pin = Ruggeduino.DIGITAL_END_PIN 
```

Reply:
I got too fed up of trying to deal with these custom messages and arrays so removed the reading LED status functionality. I'm happy for you to revert and fix it instead though